test/test_browser.py

```diff
@@ -4756,18 +4756,61 @@ def test_pthread_growth(self, emcc_args):
   def test_pthread_reltime(self):
     self.btest_exit('pthread/test_pthread_reltime.cpp', emcc_args=['-pthread', '-sPTHREAD_POOL_SIZE'])
 
-  # Tests that it is possible to load the main .js file of the application manually via a Blob URL,
+  # Tests that it is possible to load the main .js file of the application manually via mainScriptUrlOrBlob,
   # and still use pthreads.
-  def test_load_js_from_blob_with_pthreads(self):
+  @parameterized({
+    'blob': (False, True),
+    'url': (False, False),
+    'blob_es6': (True, True),
+    'url_es6': (True, False),
+  })
+  def test_mainScriptUrlOrBlob(self, es6, use_blob):
     # TODO: enable this with wasm, currently pthreads/atomics have limitations
     self.set_setting('EXIT_RUNTIME')
+    js_name = 'hello_thread_with_loader.js'
+    ext = '.js'
+    if es6:
+      ext = '.mjs'
+      self.emcc_args += ['-sEXPORT_ES6']
+      js_name = 'hello_thread_with_loader.mjs'
+    if es6 and use_blob:
+      create_file('loader.mjs', '''
+        Module['locateFile'] = (path,_prefix) => path;
+        fetch('hello_thread_with_loader.mjs').then((rsp) => rsp.blob().then((blob) => {
+          Module['mainScriptUrlOrBlob'] = blob;
+          import(URL.createObjectURL(blob)).then((fac) => fac.default(Module))
+        }));
```

Comment:
Since this is ES6 I guess you could use `await` there to make this a little more readable?   Doesn't make much difference I guess?

Reply:
Nice.